pkg/data/postgres.go

```diff
@@ -0,0 +1,379 @@
+package data
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	api "kubedb.dev/apimachinery/apis/kubedb/v1alpha2"
+	cs "kubedb.dev/apimachinery/client/clientset/versioned"
+	"kubedb.dev/cli/pkg/lib"
+	"log"
+	"os"
+
+	"github.com/spf13/cobra"
+	shell "gomodules.xyz/go-sh"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+const (
+	pgCaFile   = "/tmp/root.crt"
+	pgCertFile = "/tmp/postgresql.crt"
+	pgKeyFile  = "/tmp/postgresql.key"
+)
+
+func InsertPostgresDataCMD(f cmdutil.Factory) *cobra.Command {
+	var (
+		dbName         string
+		postgresDBName string
+		command        string
+		rows           int
+	)
+
+	pgInsertCmd := &cobra.Command{
+		Use: "postgres",
+		Aliases: []string{
+			"postgresql",
+			"pgsql",
+			"pg",
+		},
+		Short: "Insert data to a postgres object's pod",
+		Long:  `Use this cmd to insert data into a postgres object's primary pod.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				log.Fatal("Enter postgres object's name as an argument. Your commands will be applied on a database inside it's primary pod")
+			}
+			dbName = args[0]
+
+			namespace, _, err := f.ToRawKubeConfigLoader().Namespace()
+			if err != nil {
+				klog.Error(err, "failed to get current namespace")
+			}
+
+			opts, err := newPostgresOpts(f, dbName, namespace, postgresDBName)
+			if err != nil {
+				log.Fatalln(err)
+			}
+
+			if rows <= 0 {
+				log.Fatal("Inserted rows must be greater than 0")
+			}
+
+			tunnel, err := lib.TunnelToDBService(opts.config, dbName, namespace, api.PostgresDatabasePort)
+			if err != nil {
+				log.Fatal("couldn't creat tunnel, error: ", err)
+			}
+
+			if rows > 0 {
+				command = `
+						DO $$ BEGIN
+							IF NOT EXISTS (SELECT 1 FROM information_schema.tables WHERE table_schema = 'public' AND table_name = 'appscode_kubedb_postgres_test_table') THEN
+								CREATE TABLE appscode_kubedb_postgres_test_table (value int not null);
+							END IF;
+						END $$;
+                ` + "\n" + fmt.Sprintf("INSERT INTO appscode_kubedb_postgres_test_table (value) values (generate_series(1,%v));", rows)
+				err = opts.executeCommand(tunnel.Local, command)
+				if err != nil {
```

Comment:
You can write separate functions for insert, verify and delete and initiate the command there. Then call executeCommand function from there.

Reply:
Done.